File: DSA Practice/LeetCode/2486. Append Characters to String to Make Subsequence/solution.java
class Solution {
    public int appendCharacters(String s, String t) {
        char[] schar = s.toCharArray();
        char[] tchar = t.toCharArray();

        int tpointer = 0;

        for (char c : schar){
            if (tpointer == tchar.length){
                break;
            }
            if (c == tchar[tpointer]){
                tpointer++;
            }
        }

        return tchar.length - tpointer;
    }
}

File: DSA Practice/LeetCode/2486. Append Characters to String to Make Subsequence/solution.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        tpointer = 0
        
        for i in s:
            if tpointer == len(t):
                break
            if i == t[tpointer]:
                tpointer += 1
        
        return len(t) - tpointer